README.md

```diff
@@ -30,6 +30,24 @@ $ shortlinks --listen :8081 --db file:shortlinks.db
 
 Then navigate to `http://localhost:8081` and create your first shortlink!
 
+## Variables
+
+In addition to simple links, variables can be added to a link to be filled out based on what the input URL is.
+
+Variables are added with `%s` in the URL. 
+For instance if your link is defined as `https://atlassian.net/browse/%s` and `j` pointed to it,
+then `s.frew.co/j` would resolve to `https://atlassian.net/browse/` and `s.few.co/j/JIRA-000` would resolve to `https://atlassian.net/browse/JIRA-000`.
+
+This works with multiple variables in a left to right fashion.
+If you define `https://some.url/q=%s&q=%s` with `some` pointed to it,
+then `s.frew.co/some/firstquery/secondquery` would resolve to `https://some.url/q=firstquery&q=secondquery`.
+
+Siumilarly, extra variables are appended to the end of the url with slashes seperating them.
```

Comment:
`Siumilarly`.

Reply:
I have a few other thoughts on this.

1. Do we actually have any use cases where people would need more than a single parameter?  If not I'd rather keep it simple and only allow a single parameter.
2. Are you aware that in URLs the parameter part (q=...) and the path part (/foo/bar) actually have different encodings, and thus you will likely be unable to generate certain urls at all with this.

Reply:
Anything thats a cluster specific would probably need 2 parameters since one would be setting the cluster and the other would set whatever the dynamic parameter is.

And to the second point, yes, there are certain URLs that will not be possible to craft with these without introducing more complexity however given 2 of the 3 other popular link shortening tools behave the same in this regard, it seems reasonable to have this limitation. If it becomes an issue we can support something more fully featured like full go templates like talescale's golinks.

Reply:
Can you give an example of one of these clustered endpoints?